### DISCUSSION TITLE
Editor Play

### DISCUSSION THREAD

**Discussion by wtmsuperman** [KyryloKuzyk/PrimeTween#62] [Ideas]
Gread job! For So Many Years,finally a **PERFECT** solution for tween!

I can't wait to buy PrimeTweenPro!
Goodbye DotweenPro,You are not Pro at all.

But can you first add a feature of playing Tween in the editor right now?No need too Pro,just play the tween,all I need is a function like EditorPreviewManager.Play(tween)

**Comment by KyryloKuzyk** [comment]
Huge thanks! I hope you have a good time using PrimeTween :)

The 'Edit Mode' execution feature is on my radar and it will be available in Pro version as a part of a feature that will let users create animations in Inspector without the need to code. The way I see it, playing tweens in Editor should not be different from playing them at runtime, so an additional `EditorPreviewManager.Play()` method will not be needed.

However, I'm not yet sure whether the 'Edit Mode' execution feature will end up in the free version or not.

**Comment by o2co2** [comment]
Hope to have as soon as possible

**Comment by o2co2** [comment]
When will it support?

**Comment by octia** [comment]
Before the pro version comes out, is there a workaround to create a 0-duration sequence that can be executed in edit mode? 
My goal is to preview the start/end states of a sequence, when modifying a prefab. 
I'm using a custom inspector, so simple mySequence.SetToFirstFrame() and mySequence.SetToLastFrame() functions would be enough.

**Comment by HikoRuwk** [comment]
A temporary solution would be create a wrapper for Tween/Sequence and cached start value and end valud into tween setting. 
Then create a editor ticker to sample the Tween/Sequence by yourself :D

**Comment by octia** [comment]
That's a huge pain, considering how much work pretty much replicating the code for the tweens and sequences would have to go into this wrapper, to retain all functionality - especially if you're using custom tweens, like I am. I mean, the start and end values aren't exposed in any way, so you'd have to make a wrapper for both, with all of the constructors and functions like chain and group implemented. With access to the source code, this _should_ be pretty quick. But since it's not public, well...

**Reply by HikoRuwk** [reply]
You can get the source code. Once you download the package, you can get it in package cache(in Library folder). Since you can get the source code you can also give a try. I think a solution would be have another editor ticker inside his PrimeTweenManager in editor time for previewing tweens.

I do create a simple wrapper and have a try(without source code), and I can only preivew some basic tweens and as I choose to cache start&end values in my tween wrapper. I agree it is a lot of work to make it work without source code, but if you just gonna use it for some simple UI animations, you dun really need to do that much :)

**Comment by KyryloKuzyk** [comment]
@wtmsuperman @o2co2 @octia @HikoRuwk @Alex-Rachel
I added the support for **Edit mode** animations in version **1.3.1**! Now you can preview animations and use `PrimeTween.Delay()` in Edit mode the same way it's used at runtime.

I also updated the Demo project to show how to preview animations without playing the scene: every animation in the Demo project now has the 'Play Animation' button in its Inspector window.
